common/rya.api/src/main/java/org/apache/rya/api/client/LoadStatementsFile.java

```diff
@@ -31,14 +31,16 @@
 @DefaultAnnotation(NonNull.class)
 public interface LoadStatementsFile {
 
-    /**
-     * Loads a local file of RDF statements into an instance of Rya.
-     *
-     * @param ryaInstanceName - The name of the Rya instance the statements will be loaded into. (not null)
-     * @param statementsFile - A file that holds RDF statements that will be loaded. (not null)
-     * @param format - The format of the statements file. (not null)
-     * @throws InstanceDoesNotExistException No instance of Rya exists for the provided name.
-     * @throws RyaClientException Something caused the command to fail.
-     */
-    public void loadStatements(String ryaInstanceName, Path statementsFile, RDFFormat format) throws InstanceDoesNotExistException, RyaClientException;
+	/**
+	 * Loads a local file of RDF statements into an instance of Rya.
+	 *
+	 * @param ryaInstanceName - The name of the Rya instance the statements will be loaded into. (not null)
+	 * @param statementsFile - A file that holds RDF statements that will be loaded. (not null)
+	 * @param format - The format of the statements file. (not null)
+	 * @param flushEachUpdate - If true, each statement will be flushed individually. If false, statements will be
+	 *            grouped in an unspecified manner. A value of false will typically yield higher ingest rates.
+	 * @throws InstanceDoesNotExistException No instance of Rya exists for the provided name.
+	 * @throws RyaClientException Something caused the command to fail.
+	 */
+    public void loadStatements(String ryaInstanceName, Path statementsFile, RDFFormat format, boolean flushEachUpdate) throws InstanceDoesNotExistException, RyaClientException;
```

Comment:
I can see value in exposing the flush parameter and other parameters to the user that are specific to the execution instance (as opposed to the particular connection instance).  In the case of the flush parameter, a user might be more concerned with higher ingest and less concerned with having the table accurately reflect the data during ingest.  On the other hand, when ingesting another file, they might want the table to immediately represent their data.  In general, I think that there are a number of execution instance parameters that a user should be able to specify for some of the shell commands.  While issuing a query through the shell, the user should be able to indicate which indexers they want to use and whether or not they want to use PCJs.  These are all execution specific parameters.  Unfortunately, the current API does not support this, and a user has to reconnect each time they want to change their query configuration or ingest configuration.  Might be best to write up a Jira ticket for this and make it a wish for a new feature.